morecantile/models.py

```diff
@@ -726,19 +744,27 @@ def tiles(
             zooms = (zooms,)
 
         if truncate:
-            west, south = truncate_lnglat(west, south)
-            east, north = truncate_lnglat(east, north)
+            west, south = self.truncate_lnglat(west, south)
+            east, north = self.truncate_lnglat(east, north)
 
         if west > east:
-            bbox_west = (-180.0, south, east, north)
-            bbox_east = (west, south, 180.0, north)
+            bbox_west = (self.bbox.left, south, east, north)
+            bbox_east = (west, south, self.bbox.right, north)
             bboxes = [bbox_west, bbox_east]
         else:
             bboxes = [(west, south, east, north)]
 
         for w, s, e, n in bboxes:
+            # Clamp bounding values.
+            w = max(self.bbox.left, w)
+            s = max(self.bbox.bottom, s)
+            e = min(self.bbox.right, e)
+            n = min(self.bbox.top, n)
```

Comment:
This is the fix for #76 (it was in mercantile but not in morecantile)
https://github.com/mapbox/mercantile/blob/fe3762d14001ca400caf7462f59433b906fc25bd/mercantile/__init__.py#L531-L536